File: lib/reports/theme.ts
/**
 * Tokens de marca do relatório — espelham app/globals.css (paleta Suricatoos)
 * para dar identidade coesa com a plataforma (melhor que o tema stock do Office
 * dos modelos de referência). Hex SEM '#' para consumo direto pelo renderer
 * Python (RGBColor.from_string / reportlab) via JSON.
 */
import type { Severity } from "./report-model";

export const REPORT_THEME = {
  primary: "2456E6",
  brandCoral: "FF7678",
  ink: "0E1B2E",
  muted: "5B6B84",
  surface: "FFFFFF",
  surfaceAlt: "F4F7FC",
  border: "E2E8F5",
  success: "12996B",
  warning: "C9820B",
  destructive: "DA2C3C",
  chart: ["2456E6", "3F6EF0", "6F9BF5", "12996B", "C9820B"],
  fonts: { display: "Archivo", body: "Geist", mono: "Geist Mono" },
} as const;

export const SEVERITY_COLOR: Record<Severity, string> = {
  critical: "DA2C3C",
  high: "E8590C",
  medium: "C9820B",
  low: "2456E6",
  info: "5B6B84",
};

File: packages/report-renderer/theme.py
"""Tokens de marca do relatorio (espelham lib/reports/theme.ts / app globals.css).

Hex SEM '#'. Fontes: usa-se Calibri (presente em Word/PowerPoint) para body e
titulos por seguranca de renderizacao no cliente; a marca vem sobretudo da
PALETA (azul/coral) e do layout. Fontes proprias (Archivo/Geist) podem ser
embutidas depois.
"""

PRIMARY = "2456E6"
CORAL = "FF7678"
INK = "0E1B2E"
MUTED = "5B6B84"
SURFACE = "FFFFFF"
SURFACE_ALT = "F4F7FC"
BORDER = "E2E8F5"
SUCCESS = "12996B"
WARNING = "C9820B"
DESTRUCTIVE = "DA2C3C"

CHART = ["2456E6", "3F6EF0", "6F9BF5", "12996B", "C9820B"]

SEVERITY_COLOR = {
    "critical": "DA2C3C",
    "high": "E8590C",
    "medium": "C9820B",
    "low": "2456E6",
    "info": "5B6B84",
}
SEVERITY_LABEL = {
    "critical": "Critico",
    "high": "Alto",
    "medium": "Medio",
    "low": "Baixo",
    "info": "Informativo",
}
SEVERITY_ORDER = ["critical", "high", "medium", "low", "info"]

CALLOUT_COLOR = {
    "info": PRIMARY,
    "warning": WARNING,
    "critical": DESTRUCTIVE,
    "success": SUCCESS,
}

FONT_DISPLAY = "Calibri"
FONT_BODY = "Calibri"

WORDMARK = "Suricatoos"
